### ISSUE TITLE
more than one song

### ISSUE THREAD

**Issue by masterofcaribou** [TechShivvy/spicetify-extensions#3]
hi, love this feature! is there any way to get it to keep playing random songs after the first one? i'd be interested to just keep listening :)

**Comment by TechShivvy** [comment]
Glad you love this feature!

If you have the auto-play feature on in your Spotify settings, it would continue playing songs (not so randomly!?).

Ill definitely add this feature in the future though.

**Comment by fourteen-1** [comment]
came to request exactly this. +1. Add this much needed feature, please dev!

**Comment by wooziwoozi** [comment]
I was coming here to ask this as well. I would love it to generate a 10-20 song playlist of the most random stuff.

**Comment by TechShivvy** [comment]
Hey, me again, just a quick note: my friend(@Hitesh1090) and I actually hacked together a rough version of this on the same day. It’s not super polished, but it works - we added a button to toggle autoplay for random songs. Still a few edge cases to sort out, but overall it does the job. I’ll clean it up a bit over the next few days. Props to @Hitesh1090 for jumping in and helping out!

cc: @wooziwoozi /@fourteen-1 /@masterofcaribou

**Comment by fourteen-1** [comment]
@TechShivvy Is this update rolled out on the spicetify marketplace because it doesn't show on my end. If not, from where would I get it? Thanks.

**Comment by TechShivvy** [comment]
@fourteen-1 yeah it's already rolled out. try re-installing once or it might take some time for you. at my end also, it took time for the changes to be reflected in the marketplace.

**Comment by fourteen-1** [comment]
@TechShivvy Okay I got the update. But the autoplay function isn't working. 

- When spotify's native autoplay is toggled off, the random track is played to completion and playback is stopped.
- When spotify's native autoplay is toggled on, the random track is played and then similar genre tracks fill up the queue and they are played (this is spotify's default autoplay nature, I believe).

**Comment by TechShivvy** [comment]
ill check it

**Comment by TechShivvy** [comment]
Hey all, the extension was broken and I fixed it recently. (autoplay feature works)

**Fixed:**
- Extension is fully working again - the APIs it relied on were deprecated/broken, everything has been migrated to the new ones
- Autoplay no longer abruptly cuts the current song - transitions are seamless now
- Fixed a bug where playlists under 50 were never actually randomised
- No more infinite loading if a track isn't playable - retries are bounded now

**What's new:**
- **Right-click any user profile** to play/queue a random track from their playlists, or set them as your randomiser profile
- **Right-click the topbar button** to open a settings modal with autoplay toggle, profile URI input (accepts Spotify share links now!), and keyboard shortcuts
- **Settings are saved** - autoplay state and profile URI persist across restarts
- **Keyboard shortcuts** - `Alt+R` (play random), `Alt+A` (toggle autoplay), `Alt+E` (open settings)

**UI change:**
The 3 topbar buttons are now 1 - left-click plays a random song, right-click opens settings. Same functionality, cleaner topbar.

Thanks for the patience, and sorry for the wait! If this works well for you, a ⭐ on the repo would mean a lot :)

Latest version: [v1.2.0](https://github.com/TechShivvy/spicetify-extensions/tree/v1.2.0) / PR: #9 

cc: @wooziwoozi / @fourteen-1 / @masterofcaribou

**Comment by TechShivvy** [comment]
> I would love it to generate a 10-20 song playlist of the most random stuff.

will note it down and try to include it in the next release!

cc: @masterofcaribou
